switchapi/es2k/switch_lag.c

```diff
@@ -0,0 +1,472 @@
+        "error: %s\n",
+        device, switch_error_to_string(status));
+    return status;
+  }
+
+  *lag_h = switch_lag_handle_create(device);
+  CHECK_RET(*lag_h == SWITCH_API_INVALID_HANDLE, SWITCH_STATUS_NO_MEMORY);
+
+  status = switch_lag_get(device, *lag_h, &lag_info);
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+  lag_info->lag_handle = *lag_h;
+
+  status = SWITCH_LIST_INIT(&(lag_info->lag_members));
+  SWITCH_ASSERT(status == SWITCH_STATUS_SUCCESS);
+
+  SWITCH_MEMCPY(&lag_info->api_lag_info, api_lag_info,
+                sizeof(switch_api_lag_info_t));
+
+  return status;
+}
+
+/*
+ * Routine Description:
+ *   @brief Delete LAG
+ *
+ * Arguments:
+ *   @param[in] device - device
+ *   @param[out] lag_h - LAG handle
+ *
+ * Return Values:
+ *    @return  SWITCH_STATUS_SUCCESS on success
+ *             Failure status code on error
+ */
+switch_status_t switch_api_lag_delete(switch_device_t device,
+                                      switch_handle_t lag_h) {
+  switch_status_t status = SWITCH_STATUS_SUCCESS;
+  switch_lag_info_t* lag_info = NULL;
+
+  status = switch_lag_get(device, lag_h, &lag_info);
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+  //--------------------- Tx Path : Del Case ----------------------//
+  status = switch_pd_tx_lag_table_entry(device, lag_info, false);
+  if (status != SWITCH_STATUS_SUCCESS) {
+    krnlmon_log_error(
+        "Failed to delete tx lag table entry on device %d: "
+        ",error: %s\n",
+        device, switch_error_to_string(status));
+    return status;
+  }
+
+  //--------------------- Rx Path : Del Case ----------------------//
+  status = switch_pd_rx_lag_table_entry(device, lag_info, false);
+  if (status != SWITCH_STATUS_SUCCESS) {
+    krnlmon_log_error(
+        "Failed to delete rx lag table entry on device %d: "
+        ",error: %s\n",
+        device, switch_error_to_string(status));
+    return status;
+  }
+
+  status = switch_lag_handle_delete(device, lag_h);
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+  return status;
+}
+
+/*
+ * Routine Description:
+ *   @brief Create LAG member
+ *
+ * Arguments:
+ *   @param[in] device - device
+ *   @param[in] api_lag_member_info - LAG member info
+ *   @param[out] lag_member_h - LAG member handle
+ *
+ * Return Values:
+ *    @return  SWITCH_STATUS_SUCCESS on success
+ *             Failure status code on error
+ */
+switch_status_t switch_api_lag_member_create(
+    switch_device_t device, switch_api_lag_member_info_t* api_lag_member_info,
+    switch_handle_t* lag_member_h) {
+  switch_status_t status = SWITCH_STATUS_SUCCESS;
+  switch_lag_member_info_t* lag_member_info = NULL;
+
+  *lag_member_h = switch_lag_member_handle_create(device);
+  CHECK_RET(*lag_member_h == SWITCH_API_INVALID_HANDLE,
+            SWITCH_STATUS_NO_MEMORY);
+
+  status = switch_lag_member_get(device, *lag_member_h, &lag_member_info);
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
```

Comment:
lag_member_h created above at line:195 needs to be deleted before returning here?

Reply:
Addressed.